lbrynet/core/LBRYMetadata.py

```diff
@@ -0,0 +1,135 @@
+import requests
+import json
+import time
+
+from copy import deepcopy
+from googlefinance import getQuotes
+from lbrynet.conf import CURRENCIES
+import logging
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.INFO)
+
+BITTREX_FEE = 0.0025
+
+SOURCE_TYPES = ['lbry_sd_hash', 'url', 'btih']
+BASE_METADATA_FIELDS = ['title', 'description', 'author', 'language', 'license', 'content-type', 'sources']
+OPTIONAL_METADATA_FIELDS = ['thumbnail', 'preview', 'fee', 'contact', 'pubkey']
+
+#v0.0.1 metadata
+METADATA_REVISIONS = {'0.0.1': {'required': BASE_METADATA_FIELDS, 'optional': OPTIONAL_METADATA_FIELDS}}
+#v0.0.2 metadata additions
+METADATA_REVISIONS['0.0.2'] = {'required': ['nsfw', 'ver'], 'optional': ['license_url']}
+CURRENT_METADATA_VERSION = '0.0.2'
+
+
+#v0.0.1 fee
+FEE_REVISIONS = {'0.0.1': {'required': ['amount', 'address'], 'optional': []}}
+CURRENT_FEE_REVISION = '0.0.1'
+
+
+class LBRYFeeFormat(dict):
+    def __init__(self, fee_dict):
+        dict.__init__(self)
+        self.fee_version = None
+        f = deepcopy(fee_dict)
+        assert len(fee_dict) == 1
+        for currency in fee_dict:
+            assert currency in CURRENCIES, "Unsupported currency: %s" % str(currency)
```

Comment:
I was doing str for in the case that someone manually posts a malformed fee, so if the assert errors the error message itself can't error